### ISSUE TITLE
SameDiff: NPE instead of missing placeholder exception in test case

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/deeplearning4j#8648]
```
        SameDiff sd = SameDiff.create();

        int nOut = 4;
        int minibatch = 10;
        SDVariable predictions = sd.var("in", DataType.DOUBLE, minibatch, nOut);
        SDVariable labels = sd.var("labels", DataType.DOUBLE, -1, nOut);

        INDArray wpe = Nd4j.createFromArray(new double[]{0,0,1,1,2,2,3,3,4,4});
        SDVariable  w = sd.var("weights", Nd4j.create(new double[][]{
                {0,0,0,0}, {0,0,1,1}, {1,1,0,0}, {1,1,1,1}, {1,1,1,1},
                {2,2,2,2}, {2,2,2,2}, {2,2,2,2}, {2,2,2,2}, {2,2,2,2}}));

        LossReduce reduction = LossReduce.MEAN_BY_NONZERO_WEIGHT_COUNT;

        INDArray predictionsArr = Nd4j.randn(DataType.DOUBLE, minibatch, nOut);
        INDArray labelsArr = Nd4j.randn(DataType.DOUBLE, minibatch, nOut);

        INDArray expOut = Transforms.abs(predictionsArr.sub(labelsArr));
        SDVariable   loss = sd.loss().absoluteDifference("loss", labels, predictions, w, reduction);


        //This does not seem to actually do anything.
        TestCase tc = new TestCase(sd)
                .expectedOutput("loss", expOut);
        OpValidation.validate(tc);
```

The labels placeholder isn't provided in the test case; we should get an exception saying there's a missing placeholder (and identify which one is missing by name), instead of getting a NPE.
